prem_utils/connectors/base.py

```diff
@@ -52,7 +52,7 @@ def get_number_of_tokens_request(self, messages: list):
 
     def get_number_of_tokens_response(self, text: str):
         encoding = tiktoken.get_encoding("cl100k_base")
-        return len(encoding.encode(text))
+        return len(encoding.encode(text, disallowed_special=()))
```

Comment:
what's this for?

Reply:
If a text being tokenized by tiktoken contains a special token like `<|endoftext|>`, we will see the error:
```
ValueError: Encountered text corresponding to disallowed special token '<|endoftext|>'.
If you want this text to be encoded as a special token, pass it to `allowed_special`, e.g. `allowed_special={'<|endoftext|>', ...}`.
If you want this text to be encoded as normal text, disable the check for this token by passing `disallowed_special=(enc.special_tokens_set - {'<|endoftext|>'})`.
To disable this check for all special tokens, pass `disallowed_special=()`.
```

I observed that one of the SLMs generated the `<|endoftext|>` token. Hence to deal with it, I added this. It doesn't impact other tokens.